compiler/src/Language/Bond/Codegen/Java/Class_java.hs

```diff
@@ -10,25 +10,216 @@ module Language.Bond.Codegen.Java.Class_java
+        javaStructFieldDescriptorInitTypeExpr (BT_TypeParam param) = [lt|#{paramName param}|]
+        javaStructFieldDescriptorInitTypeExpr (BT_UserDefined e@Enum {..} _) = [lt|#{qualifiedDeclaredTypeName java e}.BOND_TYPE|]
+        javaStructFieldDescriptorInitTypeExpr (BT_UserDefined s@Struct {..} params) = [lt|#{structFieldDescriptorInitStructExpr java (qualifiedDeclaredTypeName java s) params}|]
+        javaStructFieldDescriptorInitTypeExpr (BT_UserDefined s@Forward {..} params) = [lt|#{structFieldDescriptorInitStructExpr java (qualifiedDeclaredTypeName java s) params}|]
+        javaStructFieldDescriptorInitTypeExpr t = error $ "invalid declaration type for javaFieldBondTypeExpression: " ++ (show t)
+
+
+-- returns an type descriptor expression for a given struct base
+structBaseDescriptorInitStructExpr :: MappingContext -> (Maybe Type) -> Text
+structBaseDescriptorInitStructExpr _ Nothing = [lt|null|]
+structBaseDescriptorInitStructExpr java (Just t) = structFieldDescriptorInitTypeExpr java t
+
+
+-- returns an expression for the default value of a field that has an explicit default, used in intiialization of struct field descriptors
+fieldDefaultValueInitParamExpr :: MappingContext -> Type -> Maybe Default -> Text
+fieldDefaultValueInitParamExpr java type_ default_ = javaFieldDefaultValueInitParamExpr type_ default_
+    where
+        javaFieldDefaultValueInitParamExpr _ (Just (DefaultBool val)) = if val
+            then [lt|true|]
+            else [lt|false|]
+        javaFieldDefaultValueInitParamExpr BT_Int8 (Just (DefaultInteger val)) = [lt|(byte)#{val}|]
+        javaFieldDefaultValueInitParamExpr BT_Int16 (Just (DefaultInteger val)) = [lt|(short)#{val}|]
+        javaFieldDefaultValueInitParamExpr BT_Int32 (Just (DefaultInteger val)) = [lt|#{val}|]
+        javaFieldDefaultValueInitParamExpr BT_Int64 (Just (DefaultInteger val)) = [lt|#{val}L|]
+        javaFieldDefaultValueInitParamExpr BT_UInt8 (Just (DefaultInteger val)) = [lt|(byte)#{val}|]
+        javaFieldDefaultValueInitParamExpr BT_UInt16 (Just (DefaultInteger val)) = [lt|(short)#{val}|]
+        javaFieldDefaultValueInitParamExpr BT_UInt32 (Just (DefaultInteger val)) = [lt|#{val}|]
+        javaFieldDefaultValueInitParamExpr BT_UInt64 (Just (DefaultInteger val)) = [lt|#{val}L|]
+        javaFieldDefaultValueInitParamExpr BT_Float (Just (DefaultFloat val)) = [lt|#{val}F|]
+        javaFieldDefaultValueInitParamExpr BT_Double (Just (DefaultFloat val)) = [lt|#{val}D|]
+        javaFieldDefaultValueInitParamExpr BT_String (Just (DefaultString val)) = [lt|"#{val}"|]
+        javaFieldDefaultValueInitParamExpr BT_WString (Just (DefaultString val)) = [lt|"#{val}"|]
+        javaFieldDefaultValueInitParamExpr (BT_UserDefined e@Enum {..} _) (Just (DefaultEnum val)) = [lt|#{qualifiedDeclaredTypeName java e}.#{val}|]
+        javaFieldDefaultValueInitParamExpr t _ = error $ "invalid declaration type for fieldDefaultValueInitParamExpr: " ++ (show t)
```

Comment:
With the code in this PR, building compat fails for me with
`gbc: invalid declaration type for fieldDefaultValueInitParamExpr: BT_Double`
Presumably, a `BT_Double None` is making it through to this function. You mentioned running it successfully, so maybe you've already fixed this.